src/i18n.c

```diff
@@ -2,11 +2,12 @@
 #include <string.h>
 #include <stdlib.h>
 
-#define i18n_map_size 3
-char *i18n_map[i18n_map_size][4] = {
-    {"zh_CN", "激活 ", "转到“设置”以激活 ", "。"},
-    {"zh_TW", "啟用 ", "移至[設定]以啟用 ", "。"},
-    {"zh_HK", "啟用 ", "移至[設定]以啟用 ", "。"},
+#define i18n_map_size 4
+char *i18n_map[i18n_map_size][5] = {
+    {"zh_CN", "激活 ", "", "转到“设置”以激活 ", "。"},
+    {"zh_TW", "啟用 ", "",  "移至[設定]以啟用 ", "。"},
+    {"zh_HK", "啟用 ", "", "移至[設定]以啟用 ", "。"},
+    {"ja_JP", "", " をアクティブ化", "「システム環境設定」 ", "をアクティブ化に行ってください。"},
```

Comment:
```suggestion
    {"ja_JP", SYSTEM_NAME" をアクティブ化", "「システム環境設定」 "SYSTEM_NAME"をアクティブ化に行ってください。"},
```
@onion108 with current `i18n.c` structure there is no need to suffer with string builder. Just rebase to master and add your text again :)

Also, you may add (not more than one) newline in second string if it is needed :)

Also also, I found different activation text on Japanese~

![](https://itojisan.xyz/wp-content/uploads/2018/03/46bb25d0bb1b0aa327011ea8695896ae-380x150.gif)

Reply:
ok, merges

Reply:
Thanks!